Autopilot/Inc/SensorFusion.hpp

```diff
@@ -0,0 +1,75 @@
+/**
+ * Sensor Fusion - Converts raw sensor data into human readable formatted structs
+ * Author: Lucy Gong, Dhruv Rawat, Anthony Bertnyk
+ */
+
+#include "AttitudeDatatypes.hpp"
+
+#ifndef SENSORFUSION_HPP
+#define SENSORFUSION_HPP
+
+// -1 = FAILED
+// 0 = SUCCESS
+// 1 = Old Data
+struct SFError_t{
+    int errorCode;
+};
+
+struct SFOutput_t {
+    float roll, pitch, yaw; //rad
+    float rollRate, pitchRate, yawRate; //rad/s
+    float airspeed; //m/s
+    float altitude; //m
+    float altitudeSpeed; //m/s
+    long double latitude; //Decimal degrees
+    float latitudeSpeed; //m/s
+    long double longitude; //Decimal degrees
+    float longitudeSpeed; //m/s
+};
+
+/**
+ * Initiate sensor fusion.
+ */ 
+void SF_Init(void);
+
+/**
+ * Get fused sensor data.
+ * @param[out]    output  Reference to an output struct for fused data.
+ */ 
+SFError_t SF_GetResult(SFOutput_t *output);
+
+//RTOS stuff that the cpp files need
+#ifdef TARGET_BUILD
+
+extern "C"
+{
+#include "cmsis_os.h"
+}
+
+const char SF_MAIL_Q_SIZE = 1;
```

Comment:
Can use `#define` here

Reply:
This will be chopped since no need for queue, but for reference, #define is not the best way to do things in c++, since it's not typesafe. constexpr is the name of the game for this kind of thing